components/shared/embedder/lib.rs

```diff
@@ -582,3 +586,71 @@ pub enum LoadStatus {
+pub struct Notification {
+    /// Title of the notification.
+    pub title: String,
+    /// Body string of the notification.
+    pub body: String,
+    /// An identifier tag for the notification. Notification with the same tag
+    /// can be replaced by another to avoid users' screen being filled up with similar notifications.
+    pub tag: String,
+    /// Direction of the notification. Includes the text direction and
+    /// actions' relative order if actions are displayed side by side.
+    ///
+    /// <https://notifications.spec.whatwg.org/#direction>
+    pub dir: NotificationDirection,
+    /// The language used in the notification's title, body, and the title of each its actions. [RFC 5646](https://datatracker.ietf.org/doc/html/rfc5646)
+    pub lang: String,
+    /// A boolean value indicates the notification should remain readily available
+    /// until the end user activates or dismisses the notification.
+    pub require_interaction: bool,
+    /// When `true`, indicates no sounds or vibrations should be made. When `None`,
+    /// the device's default settings should be respected.
+    pub silent: Option<bool>,
+    /// The URL of an icon. The icon will be displayed as part of the notification.
+    pub icon_url: Option<ServoUrl>,
+    /// Icon's raw image data and metadata.
+    pub icon_resource: Option<Arc<Image>>,
+    /// The URL of a badge. The badge is used when there is no enough space to display the notification,
+    /// such as on a mobile device's notification bar.
+    pub badge_url: Option<ServoUrl>,
+    /// Badge's raw image data and metadata.
+    pub badge_resource: Option<Arc<Image>>,
+    /// The URL of an image. The image will be displayed as part of the notification.
+    pub image_url: Option<ServoUrl>,
+    /// Image's raw image data and metadata.
+    pub image_resource: Option<Arc<Image>>,
+    /// Actions available for users to choose from for interacting with the notification.
+    pub actions: Vec<NotificationAction>,
+}
+
+/// Actions available for users to choose from for interacting with the notification.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct NotificationAction {
+    /// A string that identifies the action.
+    pub name: String,
+    /// The title string of the action to be shown to the user.
+    pub title: String,
+    /// The URL of an icon. The icon will be displayed with the action.
+    pub icon_url: Option<ServoUrl>,
+    /// Icon's raw image data and metadata.
+    pub icon_resource: Option<Arc<Image>>,
+}
+
+/// Direction of the notification. Includes the text direction and
+/// actions' relative order if actions are displayed side by side.
+///
+/// <https://notifications.spec.whatwg.org/#direction>
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
+pub enum NotificationDirection {
+    /// Follows default browser language setting.
+    Auto,
+    /// Left to right
+    Ltr,
+    /// Right to left
+    Rtl,
+}
```

Comment:
I would skip this property for now. Presumably, Servo should resolve `Auto` here, but we aren't doing that.